spec/models/jewelry_item_spec.rb

```diff
@@ -451,16 +451,66 @@
     let(:item) { create(:jewelry_item, name: 'Gold Diamond Ring') }
 
     context 'when there is a single matching canonical model' do
-      before do
+      let!(:matching_canonical) do
         create(
           :canonical_jewelry_item,
           :with_enchantments,
           name: 'Gold Diamond Ring',
         )
       end
 
-      it 'adds enchantments' do
-        expect(item.enchantments.length).to eq 2
+      context "when the new item doesn't have its own enchantments" do
+        let(:item) do
+          build(
+            :jewelry_item,
+            name: 'Gold diamond ring',
+          )
+        end
+
+        it 'adds enchantments from the canonical model' do
+          item.save!
+          expect(item.enchantments.length).to eq 2
+        end
+
+        it 'sets "added_automatically" to true on new associations' do
+          item.save!
+          expect(item.enchantables_enchantments.pluck(:added_automatically).uniq)
+            .to eq [true]
+        end
+
+        it 'sets the correct strengths', :aggregate_failures do
+          item.save!
+
+          matching_canonical.enchantables_enchantments.each do |join_model|
+            has_matching = item.enchantables_enchantments.any? do |model|
+              model.enchantment == join_model.enchantment && model.strength == join_model.strength
```

Comment:
Currently this will be be `nil` - we should adjust either the factory or the test setup to make sure there is actually a value here.